File: java/src/main/java/XtdbHelloWorld.java
import java.sql.DriverManager;
import java.sql.SQLException;

public class XtdbHelloWorld {

    public static void main(String[] args) {
        String host = System.getenv("XTDB_HOST");
        if (host == null || host.isEmpty()) {
            host = "xtdb";
        }
        String url = "jdbc:xtdb://" + host + ":5432/xtdb";

        try (var connection =
                     DriverManager.getConnection(url, "xtdb", "xtdb");
             var statement = connection.createStatement()) {

            // Insert records using XTDB's RECORDS syntax
            statement.execute("INSERT INTO users RECORDS {_id: 'jms', name: 'James'}, {_id: 'joe', name: 'Joe'}");

            // Query the table and print results
            try (var resultSet = statement.executeQuery("SELECT * FROM users")) {
                System.out.println("Users:");

                while (resultSet.next()) {
                    System.out.printf("  * %s: %s%n", resultSet.getString("_id"), resultSet.getString("name"));
                }
            }
        } catch (SQLException e) {
            e.printStackTrace();
        }
    }
}

File: php/XtdbHelloWorld.php
<?php

// Connect to XTDB using ext-pq (PECL pq) driver
// This gives us the ability to specify parameter OIDs for custom types
use pq\Connection;

try {
    // Connect to XTDB
    $connection = new Connection("host=xtdb port=5432 dbname=xtdb user=xtdb password=");

    // Insert records using XTDB's RECORDS syntax
    $insert_query = "INSERT INTO users RECORDS {_id: 'jms', name: 'James'}, {_id: 'joe', name: 'Joe'}";
    $connection->exec($insert_query);

    // Query the table and print results
    $result = $connection->exec("SELECT * FROM users");

    echo "Users:\n";

    while ($row = $result->fetchRow(\pq\Result::FETCH_ASSOC)) {
        echo "  * " . $row['_id'] . ": " . $row['name'] . "\n";
    }

} catch (\pq\Exception $e) {
    echo "Error: " . $e->getMessage() . "\n";
    exit(1);
}

?>
